app/addons.go

```diff
@@ -6,30 +6,44 @@ import (
 	"errors"
 
 	serrors "github.com/cyverse-de/subscriptions/errors"
+	"github.com/sirupsen/logrus"
 
 	qmsinit "github.com/cyverse-de/go-mod/pbinit/qms"
+	reqinit "github.com/cyverse-de/go-mod/pbinit/requests"
 	"github.com/cyverse-de/p/go/qms"
+	"github.com/cyverse-de/p/go/requests"
 	"github.com/cyverse-de/subscriptions/db"
 )
 
-func (a *App) AddAddonHandler(subject, reply string, request *qms.AddAddonRequest) {
-	var err error
-
-	log := log.WithField("context", "adding new available addon")
-
-	response := qmsinit.NewAddonResponse()
+func (a *App) sendAddonResponseError(reply string, log *logrus.Entry) func(context.Context, *qms.AddonResponse, error) {
+	return func(ctx context.Context, response *qms.AddonResponse, err error) {
+		log.Error(err)
+		response.Error = serrors.NatsError(ctx, err)
+		if err = a.client.Respond(ctx, reply, response); err != nil {
+			log.Error(err)
+		}
+	}
+}
 
-	sendError := func(ctx context.Context, response *qms.AddonResponse, err error) {
+func (a *App) sendAddonListResponseError(reply string, log *logrus.Entry) func(context.Context, *qms.AddonListResponse, error) {
+	return func(ctx context.Context, response *qms.AddonListResponse, err error) {
 		log.Error(err)
 		response.Error = serrors.NatsError(ctx, err)
 		if err = a.client.Respond(ctx, reply, response); err != nil {
 			log.Error(err)
 		}
 	}
+}
```

Comment:
Probably not a big deal, but it feels wasteful to have repeated the whole function body here. Given that `a.client.Respond` takes a `gotelnats.DEResponse`, would it be possible to combine these two functions and use that instead, or is there a problem or reason not to do that I'm missing?

Reply:
Yeah, Go generics are funky and I couldn't figure out a way to combine them.